### PR TITLE
Support deployment protection rule event type

### PR DESCRIPTION
**Pull request by BoyangHuang**
This commit adds the deployment_protection_rule event type to the list of parseable event types

Fixes: #2774.

### CODE REVIEW

File: github/event.go
Comment (gmlewis) [inline github/event.go:52]:
```suggestion
	case "DeploymentProtectionRuleEvent":
```

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:176]:
```suggestion
// DeploymentProtectionRuleEvent represents a deployment protection rule event.
```

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:177]:
```suggestion
// The Webhook event name is "deployment_protection_rule".
```

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:189]:
```suggestion
	PullRequests           []*PullRequest  `json:"pull_requests,omitempty"`
```

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:183]:
```suggestion
Event *string `json:"event,omitempty"`
```

File: github/messages.go
Comment (gmlewis) [inline github/messages.go:58]:
```suggestion
		"deployment_protection_rule":     "DeploymentProtectionRuleEvent",
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @BoyangHuang !
After a few tweaks, we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by BoyangHuang** [comment]
> Thank you, @BoyangHuang ! After a few tweaks, we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

Thank you for catching my mistakes :) much appreciated

**Comment by gmlewis** [comment]
Sorry, I forgot to mention that you will need to `gofmt` the files after you add my suggestions. Also make sure to run `go generate ./...` again then push the changes. Thank you!

**Review by gmlewis — APPROVED** [review]
Thank you, @BoyangHuang !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by BoyangHuang** [comment]
small bump here - is there any contributor available to give this a quick look? Thank you!

**Comment by gmlewis** [comment]
Hi @valbeat - do you have time for a code review?

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
